src/mlpack/methods/adaboost/adaboost_impl.hpp

```diff
@@ -94,9 +94,16 @@ double AdaBoost<WeakLearnerType, MatType>::Train(
       predictedLabels.n_cols);
 
   // Load the initial weights into a 2-D matrix.
-  const double initWeight = 1.0 / double(data.n_cols * numClasses);
+  //D(l)(k) is 1/2n where l is the label of the kth data point
+  //For all l other than that is 1/2n(k-1)
+  const double initWeightnotlabel = 1.0 / double(2 * data.n_cols * (numClasses - 1));
+  const double initWeightlabel = 1.0 / double(2 * data.n_cols);
   arma::mat D(numClasses, data.n_cols);
-  D.fill(initWeight);
+  D.fill(initWeightnotlabel);
+  for (size_t i = 0;i < D.n_cols;++i)
+  {
+      D.col(i)(labels(i)) = initWeightlabel;
```

Comment:
Matrix elements are accessed as (row,column).  So instead of `D.col(i)(labels(i))`, it would be clearer more efficient to use `D(labels(i), i)`.